lib/resources/authconfig.rb

```diff
@@ -0,0 +1,120 @@
+# encoding: utf-8
+# author: Matthew Dromazos
+
+require 'utils/SimpleConfig'
+require 'utils/parser'
+require 'pry'
+require 'parslet'
+
+module Inspec::Resources
+  class AuthConfig < Inspec.resource(1)
+    name 'authconfig'
+    desc 'Use the `authconfig` InSpec audit resource to test the properties set in an authconfig file.
+          The `authconfig` defines values that are used to track whether or not authentication mechanisms are enabled.'
+    example "
+    describe authconfig do
+      its ('USESSSD') { should eq 'yes' }
+    end
+
+    describe authconfig do
+      it { should be_smartcard_authen_enabled }
+    end
+
+    describe authconfig do
+      its ( 'smartcard_removal_action' ) { should eq 'Ignore' }
+    end
+    "
+
+    attr_reader :params
+
+    def initialize(host_path = nil)
+      return skip_resource 'The `authconfig` resource is not supported on your OS.' unless inspec.os.linux? || inspec.os.windows?
+      @conf_path = host_path || '/etc/sysconfig/authconfig'
+      @content = nil
+      @params = nil
+      @modules = nil
+      read_content
+    end
+
+    def installed?
+      inspec.package('authconfig').installed?
+    end
+
+    def method_missing(name)
```

Comment:
We try to avoid `method_missing` whenever possible. I thought we agreed to not parse the config file but rather expose stuff that you can get from the `authconfig` command, so this shouldn't really be needed.

Reply:
I looked at the documentation more and it shows that the config file always displays what the system is currently using.  When you make changes using the authconfig command it doesnt actually take place until you run the authconfig with the --update option, which then will write the configuration changes to the /etc/sysconfig/authconfig file.  

https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/System-Level_Authentication_Guide/authconfig-install.html